src/preprocess.ts

```diff
@@ -33,13 +34,14 @@ const getParseOptions = (isJsx: boolean): ParserOptions => ({
 		...(isJsx ? ["jsx" as const] : []),
 		["importAttributes", { deprecatedAssertSyntax: true }],
 	],
+
```

Comment:
This and the changes to `.eslintrc.cjs` in b59729c are an introduced bug. I believe they're due to `@babel/generator` re-printing the AST weirdly. This needs investigation 😕.

Reply:
Added a reduced test case in 66bf77b.

Reply:
Resolved in 122c65f. I ended up going for a hybrid approach:

* Use `@babel/generator` to print node sections that need re-printing
* Avoid re-printing other parts of the AST that are fine as-is